strawberry_django/fields/types.py

```diff
@@ -0,0 +1,159 @@
+import datetime, decimal, uuid
+import django
+import strawberry
+from django.db.models import fields
+from django.db.models.fields.reverse_related import ForeignObjectRel, ManyToOneRel
+from strawberry.arguments import UNSET
+from typing import List, Optional
+from .. import filters
+
+class auto:
```

Comment:
I like lowercase because it fits nicely with `str`, etc. But we could also consider using `AUTO` to be consistent with `UNSET`, just a thought.

Reply:
Good question. I like lower case type too. We have also discussed if `auto` is good name or if we should use some other name instead. One idea was to use `infer` instead of `auto`.